### ISSUE TITLE
something about training the haze-removal model 

### ISSUE THREAD

**Issue by ChengTianyou** [liu-vis/DualResidualNetworks#6]

**Comment by ChengTianyou** [comment]
Hi, thanks for sharing your amazing work.
And I am particularly interested in haze-removal, so I wrote the train code by myself, I trained several times, but I can't train a good model as you, could you give me some suggestions, thanks, here is my details.
First, I used the RESIDE dataset(ITS) which includes 13990 samples for training, before training, I randomly crop 256 * 256 region from an original image of size 620 * 460 as you did.
Then I use the l1 loss as you did, I set the optimizer for SGD , the lr is 0.001 and the momentum is 0.9. which you didn't mention. The batch is 16 because of the low computing resource.
Finally I trained 36 epochs, I don't thinkl training for more epochs would work, so I stopped, every two epochs I test the average loss, psnr, ssim of the SOTS, the loss and ssim decreased as epoch increased, but the psnr first increased and then decreased.The final panr was  18.08 and the ssim was 0.79, which were far lower than yours.
Above is the situations I met last weeks, and I can't figure why, do I have a big paoblems, thanks very much, looking forward for your reply.
Furthermore, the haze-removal models you used in the paper was trained in Dehaze dataset and  you didn't show the model trained in the RESIDE dataset, right?  Thanks again.

**Comment by liu-vis** [comment]
Hello, thank you for your interest in our work.
-> but I can't train a good model as you, could you give me some suggestions
Please see the followings for training details:
i) Loss = 1.1 x ssim_loss + 0.75 x L1_loss.
ii) Adam with default settings for optimizer. 
iii) learning rate = 0.0001
iv) I use batchsize = 48, I think its fine to use a smaller one (e.g., 10) if you train your model on single GPU.
v) I trained the model 2000 epochs.

-> Furthermore, the haze-removal models you used in the paper was trained in Dehaze dataset and you didn't show the model trained in the RESIDE dataset, right? 
No, the model used for Dehaze-TestA is trained using Dehaze dataset;  the model used for RESIDE-SOTS is trained using RESIDE dataset. Please see the table 6 in our paper.

**Comment by ChengTianyou** [comment]
sorry for the late reply, thanks for your suggestions, I tried and obtained a better result, thanks again!
